bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/ServiceAccountRepository.cs

```diff
@@ -84,22 +84,29 @@ public async Task<IEnumerable<Core.SecretsManager.Entities.ServiceAccount>> GetM
 
     public async Task DeleteManyByIdAsync(IEnumerable<Guid> ids)
     {
+        var targetIds = ids.ToList();
         using var scope = ServiceScopeFactory.CreateScope();
         var dbContext = GetDatabaseContext(scope);
 
+        await using var transaction = await dbContext.Database.BeginTransactionAsync();
+
```

Comment:
If you are going to add a long transaction there is a chance that it can be a deadlock victim, may want to wrap in try catch and try 3 times and if still fails to log error?

Reply:
By default, EF Core wraps any modifications preformed prior to a `await dbContext.SaveChangesAsync();` into a transaction. 
https://learn.microsoft.com/en-us/ef/core/saving/transactions#default-transaction-behavior

So previously and how EF implements `RemoveRange`

- Query for the IDs to delete
- Create a DELETE statement for each ID to remove
- Submit N number of delete statements

This would all be wrapped in one transaction. 


With the new `ExecuteDelete` [methods](https://learn.microsoft.com/en-us/ef/core/saving/execute-insert-update-delete#executedelete) it will generate a proper DELETE with a WHERE statement.

However, if you want to use multiple `ExecuteDelete` methods in one transaction, you have to explicitly declare the transaction. https://learn.microsoft.com/en-us/ef/core/saving/execute-insert-update-delete#transactions


All that to say, this won't add a new long-running transaction, it actually replaces a significantly longer running transaction.